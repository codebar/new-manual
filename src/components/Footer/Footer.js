import React from 'react';
import {
  FooterWrapper,
  FooterEmail,
  FooterSocials,
  FooterTextWrapper,
  FooterImg,
  FooterText,
  FooterLink,
} from './Footer.style';
import twitter from '../../assets/image/twitter-logo.svg';
import slack from '../../assets/image/slack-logo.svg';
import facebook from '../../assets/image/facebook-logo.svg';
import github from '../../assets/image/github-logo.svg';

const Footer = () => {
  return (
    <FooterWrapper>
      <FooterTextWrapper>
        <FooterText>Send us an email at</FooterText>
        <FooterEmail className="email" href="mailto:hello@codebar.io">
          hello@codebar.io
        </FooterEmail>
        <FooterText className="charityNo">
          Registered UK and Wales charity no. 1187776
        </FooterText>
        <FooterText>Â© 2020 codebar</FooterText>
      </FooterTextWrapper>
      <FooterSocials>
        <FooterLink href="https://slack.codebar.io/" target="_blank">
          <FooterImg src={slack} alt="Slack logo" />
        </FooterLink>
        <FooterLink href="https://github.com/codebar" target="_blank">
          <FooterImg src={github} alt="Github logo" />
        </FooterLink>
        <FooterLink href="https://twitter.com/codebar" target="_blank">
          <FooterImg src={twitter} alt="Twitter logo" />
        </FooterLink>
        <FooterLink href="https://www.facebook.com/codebarHQ" target="_blank">
          <FooterImg src={facebook} alt="Facebook logo" />
        </FooterLink>
      </FooterSocials>
    </FooterWrapper>
  );
};

export default Footer;
