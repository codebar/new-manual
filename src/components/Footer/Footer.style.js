import styled from 'styled-components';
import { Link } from 'react-router-dom';

const FooterWrapper = styled.footer`
  background-color: #3e4954;
  min-height: 35vh;
  display: flex;
  justify-content: space-between;
  padding: 2rem 4rem 0 4rem;
  margin: 0 -1rem 0 -1rem;
  color: white;
  bottom: 0;
  @media (max-width: 400px) {
    flex-direction: column;
    padding: 1rem;
  }
`;

const FooterEmail = styled.a`
  font-size: 2.15rem;
  font-weight: lighter;
  text-decoration: none;
  border-bottom: 1px solid white;
  padding: 0.1rem 0;
  color: white;
  width: 80%;
  padding-top: 0.75rem;
  @media (max-width: 400px) {
    width: 63%;
    font-size: 1.5rem;
  }
`;

const FooterTextWrapper = styled.div.attrs((props) => ({
  className: props.className,
}))`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 0.1fr 0.5fr 2.5fr 0.5fr;
  grid-template-areas: '.' '.' '.' '.';

  & .charityNo {
    padding-top: 2rem;
  }
  @media (max-width: 400px) {
    padding: 1rem;
  }
`;

const FooterText = styled.p`
  margin: 0;
`;

const FooterSocials = styled.div`
  @media (max-width: 400px) {
    padding: 1rem 0;
  }
`;
const FooterTextLink = styled(Link)`
  text-decoration: none;
  color: white;
`;

const FooterImg = styled.img`
  width: 2.5rem;
  padding: 0 1rem;
  @media (max-width: 400px) {
    width: 2rem;
  }
`;

const FooterLink = styled.a``;

export {
  FooterWrapper,
  FooterEmail,
  FooterSocials,
  FooterText,
  FooterTextWrapper,
  FooterImg,
  FooterLink,
  FooterTextLink
};
