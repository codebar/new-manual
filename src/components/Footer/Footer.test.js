import React from 'react';
import ReactDOM from 'react-dom';
import Footer from './Footer';

import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer />, div);
});

it('renders footer correctly', () => {
  render(<Footer />);
});

it('matches snapshot', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
