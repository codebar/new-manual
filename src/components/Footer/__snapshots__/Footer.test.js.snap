// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<footer
  className="sc-AxjAm eAZDYz"
>
  <div
    className="sc-AxiKw gGZwqz"
  >
    <p
      className="sc-AxhCb kqIPop"
    >
      Send us an email at
    </p>
    <a
      className="sc-AxirZ dxkyxw email"
      href="mailto:hello@codebar.io"
    >
      hello@codebar.io
    </a>
    <p
      className="sc-AxhCb kqIPop charityNo"
    >
      Registered UK and Wales charity no. 1187776
    </p>
    <p
      className="sc-AxhCb kqIPop"
    >
      Â© 2020 codebar
    </p>
  </div>
  <div
    className="sc-AxhUy hrqUPW"
  >
    <a
      className="sc-AxheI fniENP"
      href="https://slack.codebar.io/"
      target="_blank"
    >
      <img
        alt="Slack logo"
        className="sc-AxgMl deXZOn"
        src="slack-logo.svg"
      />
    </a>
    <a
      className="sc-AxheI fniENP"
      href="https://github.com/codebar"
      target="_blank"
    >
      <img
        alt="Github logo"
        className="sc-AxgMl deXZOn"
        src="github-logo.svg"
      />
    </a>
    <a
      className="sc-AxheI fniENP"
      href="https://twitter.com/codebar"
      target="_blank"
    >
      <img
        alt="Twitter logo"
        className="sc-AxgMl deXZOn"
        src="twitter-logo.svg"
      />
    </a>
    <a
      className="sc-AxheI fniENP"
      href="https://www.facebook.com/codebarHQ"
      target="_blank"
    >
      <img
        alt="Facebook logo"
        className="sc-AxgMl deXZOn"
        src="facebook-logo.svg"
      />
    </a>
  </div>
</footer>
`;
